empty-bin.lua

```diff
@@ -3,28 +3,78 @@
 -- https://gist.github.com/stonetoad/11129025
 -- http://dwarffortresswiki.org/index.php/DF2014_Talk:Bin
 
-local function moveItem(item, to_pos)
-    print('  ' .. dfhack.items.getReadableDescription(item))
-    dfhack.items.moveToGround(item, to_pos)
-end
+local argparse = require("argparse")
+
+local options, args = {
+        help = false,
+        recursive = false,
+        liquids = false
+    },
+    {...}
 
 local function emptyContainer(container)
     local items = dfhack.items.getContainedItems(container)
-
     if #items > 0 then
-        print('Emptying ' .. dfhack.items.getReadableDescription(container))
+        print("Emptying " .. dfhack.items.getReadableDescription(container))
         local pos = xyz2pos(dfhack.items.getPosition(container))
         for _, item in ipairs(items) do
-            moveItem(item, pos)
+            local skip_liquid =
+                item:getType() == df.item_type.LIQUID_MISC or
+                item:getType() == df.item_type.DRINK and not options.liquids
+            if skip_liquid then
+                print(
+                    "  " ..
+                        dfhack.items.getReadableDescription(item) ..
+                            " was skipped because the liquid flag was not provided"
+                )
+            else
+                print("  " .. dfhack.items.getReadableDescription(item))
+                dfhack.items.moveToGround(item, pos)
+                if options.recursive then
+                    emptyContainer(item)
+                end
+            end
         end
     end
 end
 
-local viewsheets = df.global.game.main_interface.view_sheets
+argparse.processArgsGetopt(
+    args,
+    {
+        {
+            "h",
+            "help",
+            handler = function()
+                options.help = true
+            end
+        },
```

Comment:
```suggestion
        {  'h', 'help',  handler = function() options.help = true end },
```
Is is customary to put short option handlers onto one line.